src/daemon/rpmostreed-os.c

```diff
@@ -1128,6 +1130,154 @@ os_handle_set_initramfs_state (RPMOSTreeOS *interface,
+                                                       kernel_args_replaced,
+                                                       kernel_args_deleted,
+                                                       kernel_arg_flags_from_options (arg_options),
+                                                       cancellable,
+                                                       &local_error);
+  if (transaction == NULL)
+    goto out;
+
+  rpmostreed_transaction_monitor_add (self->transaction_monitor, transaction);
+
+out:
+  if (local_error != NULL)
+    g_dbus_method_invocation_take_error (invocation, local_error);
+  else
+    {
+      const char *client_address = rpmostreed_transaction_get_client_address (transaction);
+      rpmostree_os_complete_kernel_args (interface, invocation, client_address);
+    }
+
+  return TRUE;
+}
+
+static gboolean
+os_handle_get_deployment_boot_config (RPMOSTreeOS *interface,
+                                      GDBusMethodInvocation *invocation,
+                                      const char *arg_deployid,
+                                      gboolean is_pending)
+{
+  glnx_unref_object OstreeSysroot *ot_sysroot = NULL;
+  GError *local_error = NULL;
+  g_autoptr(GCancellable) cancellable = NULL;
+  glnx_unref_object OstreeDeployment *target_deployment  = NULL;
+  GVariantDict boot_config_dict;
+  GVariant *boot_config_result = NULL;
+
+  /* Load the sysroot */
+  if (!rpmostreed_sysroot_load_state (rpmostreed_sysroot_get (),
+                                      cancellable,
+                                      &ot_sysroot,
+                                      NULL,
+                                      &local_error))
+    goto out;
+  const char* osname = rpmostree_os_get_name (interface);
+  if (!*arg_deployid || arg_deployid[0] == '\0')
+    {
+      if (is_pending)
+        target_deployment = rpmostree_syscore_get_origin_merge_deployment (ot_sysroot, osname);
+      else
+        target_deployment = ostree_sysroot_get_merge_deployment (ot_sysroot, osname);
+      if (target_deployment  == NULL)
+        {
+          local_error = g_error_new (G_IO_ERROR, G_IO_ERROR_FAILED,
+                                     "No deployments found for os %s", osname);
+          goto out;
+        }
+    }
+  else
+    {
+      /* If arg_deployid is specified, we ignore pending option */
+      target_deployment = rpmostreed_deployment_get_for_id (ot_sysroot, arg_deployid);
+      if (target_deployment == NULL)
+        {
+          local_error = g_error_new (RPM_OSTREED_ERROR,
+                                     RPM_OSTREED_ERROR_FAILED,
+                                     "Invalid deployment id %s",
+                                     arg_deployid);
+          goto out;
+        }
+    }
+  OstreeBootconfigParser *bootconfig = ostree_deployment_get_bootconfig (target_deployment);
+
+  /* Note because boot config is a private structure.. currently I have no good way
```

Comment:
This we should fix in libostree I think.  But it doesn't need to be done to land this change.